From 8b06e7fef50327f84140cb09a3d9237e18b38a35 Mon Sep 17 00:00:00 2001
From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
Date: Thu, 5 Sep 2024 21:14:20 +0530
Subject: [PATCH 15/16] Fix lapack larfg function signature

This patch fixes the signature of the LAPACK routine larfg. Please
see https://github.com/pyodide/pyodide/issues/3379 for more details.

Co-authored-by: Ilhan Polat <ilhanpolat@gmail.com>
Suggested-by: Hood Chatham <roberthoodchatham@gmail.com>

---
 scipy/linalg/flapack_other.pyf.src | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/scipy/linalg/flapack_other.pyf.src b/scipy/linalg/flapack_other.pyf.src
index d1a87f182..c6851fff7 100644
--- a/scipy/linalg/flapack_other.pyf.src
+++ b/scipy/linalg/flapack_other.pyf.src
@@ -2355,13 +2355,12 @@ function <prefix>lantr(norm, uplo, diag, m, n, a, lda, work) result(n2)
 
 end subroutine <prefix>lantr
 
-subroutine <prefix>larfg(n, alpha, x, incx, tau, lx)
+subroutine <prefix>larfg(n, alpha, x, incx, tau)
     integer intent(in), check(n>=1) :: n
     <ftype> intent(in,out) :: alpha
-    <ftype> intent(in,copy,out), dimension(lx) :: x
+    <ftype> intent(in,copy,out), dimension(*), depend(n,incx), check(len(x) >= (n-2)*incx) :: x
     integer intent(in), check(incx>0||incx<0) :: incx = 1
     <ftype> intent(out) :: tau
-    integer intent(hide),depend(x,n,incx),check(lx > (n-2)*incx) :: lx = len(x)
 end subroutine <prefix>larfg
 
 subroutine <prefix>larf(side,m,n,v,incv,tau,c,ldc,work,lwork)
-- 
2.39.3 (Apple Git-146)

